### PR TITLE
Add sniff to detect direct access to AI Connector API keys and give error

### LINKED ISSUE #1342: Add an error when plugins read WordPress AI Connector API keys directly
**Problem**

WordPress 7.0 stores AI provider credentials for Connectors in options following this pattern:

```text
connectors_ai_{provider}_api_key
```

Examples include:

```text
connectors_ai_openai_api_key
connectors_ai_anthropic_api_key
connectors_ai_grok_api_key
```

Plugin Check should report an error when a plugin reads any option matching this pattern directly.

**Why this check is needed**

These credentials belong to the WordPress Connectors configuration. The site owner provided them to WordPress for use through the configured connector, not for arbitrary plugin code to read directly.

Reading these options directly bypasses the AI Client / Connectors abstraction and increases the risk of exposing, logging, exporting, or misusing sensitive provider credentials.

**Proposed detection**

Use a pattern match instead of a fixed list of provider names:

```php
preg_match( '/^connectors_ai_[a-z0-9_]+_api_key$/i', $option_name )
```

Flag matching option names when used with:

```php
get_option()
get_site_option()
get_network_option()
get_options()
```

**Examples that should be flagged**

```php
get_option( 'connectors_ai_openai_api_key' );

get_site_option( 'connectors_ai_anthropic_api_key' );

get_network_option( null, 'connectors_ai_grok_api_key' );

get_options(
	array(
		'connectors_ai_openai_api_key',
		'connectors_ai_custom_provider_api_key',
	)
);
```

**Suggested message**

> Your plugin reads WordPress AI Connector API keys directly from the options table. This is not permitted.

> Options matching `connectors_ai_{provider}_api_key` store credentials configured by the site owner for WordPress Connectors. Plugins should not access these raw API keys directly. Please route AI requests through the WordPress AI Client instead, for example with `wp_ai_client_prompt()`.

### PR DESCRIPTION
**Pull request by ishitaj34**
## What?

Closes #1342 

Adds a new `AIConnectorAPIKeySniff` to detect direct access to WordPress AI Connector API key options.

## Why?

WordPress AI Connector API keys are managed by site owners and are intended to be used through the WordPress AI Client. Plugins should not read these keys directly from the options table.

## How?

- Added a new `AIConnectorAPIKeySniff` file.
- Registered the sniff in `PluginCheck/ruleset.xml`.
- Added suggested regex-based detection for option names matching:

```php
/^connectors_ai_[a-z0-9_]+_api_key$/i
```
- Added support for:
   - `get_option()`
   - `get_site_option()`
   - `get_network_option()`
   - `get_options()`

- Added unit tests covering all supported functions.

## Testing Instructions

- Check out branch.
- Navigate to the phpcs-sniffs directory.
- Run:
```
composer check-cs
```
- Run:
```
composer run-tests
```

- Verify that following examples are reported:

```php
get_option( 'connectors_ai_openai_api_key' );
get_site_option( 'connectors_ai_anthropic_api_key' );
get_network_option( null, 'connectors_ai_grok_api_key' );

get_options(
	array(
		'connectors_ai_openai_api_key',
	)
);
```

- Verify that non-matching options are not reported:

```php
get_option( 'siteurl' );
get_site_option( 'blogname' );
get_network_option( null, 'admin_email' );
```

## AI Usage Disclosure

- [ ] This PR was created without the help of AI tools
- [x] This PR includes AI-assisted code or content

If AI tools were used, please describe how they were used:

- AI assistance: Yes
- Tool(s): ChatGPT
- Used for: Discussion of implementation approaches, edge cases, test scenarios, and PR documentation. All code, tests, and final changes were reviewed, validated, and edited by author.

<a href="https://playground.wordpress.net?blueprint-url=data:application/json,%7B%22%24schema%22%3A%22https%3A%2F%2Fplayground.wordpress.net%2Fblueprint-schema.json%22%2C%22landingPage%22%3A%22%2Fwp-admin%2Ftools.php%3Fpage%3Dplugin-check%22%2C%22phpExtensionBundles%22%3A%5B%22kitchen-sink%22%5D%2C%22steps%22%3A%5B%7B%22step%22%3A%22login%22%2C%22username%22%3A%22admin%22%2C%22password%22%3A%22password%22%7D%2C%7B%22step%22%3A%22installPlugin%22%2C%22pluginZipFile%22%3A%7B%22resource%22%3A%22url%22%2C%22url%22%3A%22https%3A%2F%2Fgithub.com%2FWordPress%2Fplugin-check%2Freleases%2Fdownload%2Fci-artifacts%2Fpr-1362-2870b533a39431dbb4e0fe5f54d62380a5c25abd.zip%22%7D%2C%22options%22%3A%7B%22activate%22%3Atrue%7D%7D%5D%7D" target="_blank" rel="noopener noreferrer">
  Open WordPress Playground Preview
</a>

### CODE REVIEW

**Comment by davidperezgar** [comment]
Thanks for the PR, this looks well scoped and the detection matches the issue nicely.

One thing I’d suggest tightening before merge: `PassedParameters::getParameterFromStack()` is currently called with `array()` as the parameter-name argument, which means PHP 8 named arguments are not handled. For example, calls like:

```php
get_option( option: 'connectors_ai_openai_api_key' );
get_network_option( network_id: null, option: 'connectors_ai_grok_api_key' );
get_options( options: array( 'connectors_ai_openai_api_key' ) );
```

should also be flagged.
Could you pass the actual parameter names per function (option / options) and add named-argument test coverage?
The rest looks good to me: the sniff is narrowly targeted, the regex follows the issue, and the fixture covers the requested positional call patterns.

**Review by davidperezgar — APPROVED** [review]

**Comment by ernilambar** [comment]
Should this be warning or error?

**Comment by davidperezgar** [comment]
Yes, It could be ERROR.

File: phpcs-sniffs/PluginCheck/Sniffs/Security/AIConnectorAPIKeySniff.php [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+	/**
+	 * The group name for this group of functions.
+	 *
+	 * @var string
+	 */
+	protected $group_name = 'ai_connector_api_key';
+
+	/**
+	 * List of functions to examine.
+	 *
```
Comment (ernilambar) [inline phpcs-sniffs/PluginCheck/Sniffs/Security/AIConnectorAPIKeySniff.php:33]:
```suggestion
	 *
	 * @since 2.2.0
```

Comment (ishitaj34) [reply phpcs-sniffs/PluginCheck/Sniffs/Security/AIConnectorAPIKeySniff.php:37]: Added tag in all doc blocks.

Comment (ernilambar) [reply phpcs-sniffs/PluginCheck/Sniffs/Security/AIConnectorAPIKeySniff.php:37]: Please add in test files also.

Comment (ishitaj34) [reply phpcs-sniffs/PluginCheck/Sniffs/Security/AIConnectorAPIKeySniff.php:37]: Done.

**Review by ernilambar — APPROVED** [review]
